var/spack/repos/builtin/packages/kokkos-kernels/package.py

```diff
@@ -6,48 +6,261 @@
+    conflicts('+rdc', when='~cuda',
+              msg='Must enable CUDA to use rdc.')
+    conflicts('+enable_lambda', when='~cuda',
+              msg='Must enable CUDA to use enable_lambda.')
+
+    # Check that we haven't asked for a GPU architecture that
+    # the revision of kokkos does not support
+    conflicts('gpu_arch=Volta70', when='@:2.5.99')
+    conflicts('gpu_arch=Volta72', when='@:2.5.99')
 
-            make('build', *make_args)
+    # conflicts on kokkos version and cuda enabled
+    # see kokkos issue #1296
+    # https://github.com/kokkos/kokkos/issues/1296
+    conflicts('+cuda', when='@2.5.00:2.7.00',
+              msg='Kokkos build system has issue (#1296) when CUDA enabled'
+              ' in version 2.5.00 through 2.7.00.')
 
+    # Resources
+    # 
+    resource(
+        name='kokkos-base',
+        sha256='e0621197791ed3a381b4f02c78fa529f3cff3abb74d52157b4add17e8aa04bc4',
+        url='https://github.com/kokkos/kokkos/archive/2.9.00.tar.gz',
+        placement='kokkos-base',
+        when='@2.9.0',
+    )
+    resource(
+        name='kokkos-base',
+        sha256='1c72661f2d770517bff98837001b42b9c677d1df29f7493a1d7c008549aff630',
+        url='https://github.com/kokkos/kokkos/archive/2.8.00.tar.gz',
+        placement='kokkos-base',
+        when='@2.8.0',
+    )
+    resource(
+        name='kokkos-base',
+        sha256='01595996e612ef7410aa42fa265a23101cfe1b6993fa9810ca844db5c89ad765',
+        url='https://github.com/kokkos/kokkos/archive/2.7.00.tar.gz',
+        placement='kokkos-base',
+        when='@2.7.0',
+    )
+    resource(
+        name='kokkos-base',
+        sha256='ea232594bf746abb99ae2aafaeef5d07adc089968010a62a88aaa892106d9476',
+        url='https://github.com/kokkos/kokkos/archive/2.5.00.tar.gz',
+        placement='kokkos-base',
+        when='@2.5.0',
+    )
+
+    # Specify that v1.x is required as v2.x has API changes
+    depends_on('hwloc@:1')
+    depends_on('qthreads', when='+qthreads')
+    depends_on('cuda', when='+cuda')
+
+
+    # check wiki for more options
     def install(self, spec, prefix):
-        with working_dir('build', create=False):
-            make_args = [
-                'KOKKOSKERNELS_INSTALL_PATH=%s' % prefix,
-                'KOKKOSKERNELS_PATH=%s' % self.stage.source_path,
-                'KOKKOS_PATH=%s' % spec['kokkos'].prefix
+        generate = which(join_path(self.stage.source_path,
+                                   'scripts/generate_makefile.bash'))
+        with working_dir('build', create=True):
+            g_args = [
+                '--prefix=%s' % prefix,
+                '--with-hwloc=%s' % spec['hwloc'].prefix,
+                '--kokkos-path=%s' % join_path(self.stage.source_path, 'kokkos-base'),
+                '--kokkoskernels-path=%s' % self.stage.source_path,
+                # This reduces compile times significantly
+                #
+                '--with-scalars="float,double"',
```

Comment:
Quotes are not needed here, as this never gets run through a Bash shell, only through Python. Same below.